go/vt/tabletserver/sysloglogger/sysloglogger.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sysloglogger
+
+import (
+	"flag"
+	"fmt"
+	log "github.com/golang/glog"
+	"github.com/youtube/vitess/go/vt/servenv"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+	"log/syslog"
+)
+
+// syslogWriter is an interface that wraps syslog.Writer so it can be faked.
+type syslogWriter interface {
+	Info(string) error
+	Close() error
+}
+
+// writer holds a persistent connection to the syslog daemon
+var writer syslogWriter
+
+var logQueries = flag.Bool("log_queries", false, "Enable query logging to syslog.") // TODO
+
+func init() {
+	servenv.OnRun(func() {
+		if *logQueries {
+			var err error
+			writer, err = syslog.New(syslog.LOG_INFO, "vtquerylogger")
+			if err != nil {
+				log.Error("Query logger is unable to connect to syslog", err)
+				return
+			}
+			go run()
+		}
+	})
+}
+
+// Run logs queries to syslog, if the "log_queries" flag is set to true when starting Vitess
+func run() {
+	defer writer.Close()
+
+	ch := tabletserver.StatsLogger.Subscribe("gwslog")
+	defer tabletserver.StatsLogger.Unsubscribe(ch)
+
+	formatParams := map[string][]string{"full": {}}
+	for out := range ch {
+		stats, ok := out.(*tabletserver.LogStats)
+		if !ok {
+			err := fmt.Errorf("Unexpected value in query logs: %#v (expecting value of type %T)", out, &tabletserver.LogStats{})
+			log.Error(err)
+			continue
+		}
+		writer.Info(stats.Format(formatParams))
```

Comment:
Looking at the library code, the string is just passed down all the way to the underlying call. I don't see any weird formatting on the way...